### PR TITLE
Cgroup config

### PR DESCRIPTION
**Pull request by MykolaSuperman**

### CODE REVIEW

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

File: configs/rcar4_xen_defconfig [hunk 1]
```diff
@@ -18,7 +18,7 @@ CONFIG_SYS_BARGSIZE=1024
 CONFIG_SUPPORT_RAW_INITRD=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="root=/dev/sda4 ro rootwait console=hvc0 clk_ignore_unused pci=pcie_bus_perf vardir.disk=/dev/sda5 opendisk.target=/dev/sda6 opendisk.pkcs11=optee aosupdate.disk=/dev/aosvg/workdirs aosupdate.path=um/update_rootfs aosupdate.selinux_module=/usr/share/selinux/aos/base.pp"
+CONFIG_BOOTARGS="cgroup_no_v1=all systemd.unified_cgroup_hierarchy=1 root=/dev/sda4 ro rootwait console=hvc0 clk_ignore_unused pci=pcie_bus_perf vardir.disk=/dev/sda5 opendisk.target=/dev/sda6 opendisk.pkcs11=optee aosupdate.disk=/dev/aosvg/workdirs aosupdate.path=um/update_rootfs aosupdate.selinux_module=/usr/share/selinux/aos/base.pp"
```
Comment (al1img) [inline configs/rcar4_xen_defconfig]: Put it at the end.

File: configs/rcar3_xen_defconfig [hunk 2]
```diff
@@ -18,7 +18,7 @@ CONFIG_SYS_BARGSIZE=1024
 CONFIG_SUPPORT_RAW_INITRD=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="root=/dev/mmcblk1p4 ro rootwait console=hvc0 clk_ignore_unused pci=pcie_bus_perf vardir.disk=/dev/mmcblk1p5 opendisk.target=/dev/mmcblk1p6 opendisk.pkcs11=optee aosupdate.disk=/dev/aosvg/workdirs aosupdate.path=um/update_rootfs aosupdate.selinux_module=/usr/share/selinux/aos/base.pp"
+CONFIG_BOOTARGS="cgroup_no_v1=all systemd.unified_cgroup_hierarchy=1 root=/dev/mmcblk1p4 ro rootwait console=hvc0 clk_ignore_unused pci=pcie_bus_perf vardir.disk=/dev/mmcblk1p5 opendisk.target=/dev/mmcblk1p6 opendisk.pkcs11=optee aosupdate.disk=/dev/aosvg/workdirs aosupdate.path=um/update_rootfs aosupdate.selinux_module=/usr/share/selinux/aos/base.pp"
```
Comment (al1img) [inline configs/rcar3_xen_defconfig]: Put it at the end for line

File: configs/rcar4_xen_defconfig (hunk 1 above)
Comment (MykolaSuperman) [reply configs/rcar4_xen_defconfig]: Done

File: configs/rcar3_xen_defconfig (hunk 2 above)
Comment (MykolaSuperman) [reply configs/rcar3_xen_defconfig]: Done

**Review by al1img — APPROVED** [review]

**Comment by lorc** [comment]
Why are you trying to modify linux kernel boot arguments via the u-boot default env? Isn't there a better way?

**Comment by MykolaSuperman** [comment]
> Why are you trying to modify linux kernel boot arguments via the u-boot default env? Isn't there a better way?

@lorc I planned to do this through CONFIG_CMDLINE in the Linux configuration, but unfortunately, it did not apply and only works through u-boot bootargs

**Comment by lorc** [comment]
Probably because you missed `CONFIG_CMDLINE_FORCE`. But anyways, this is not a proper solution anyways.
It should be configured via `xenlib`. 

@Deedone  is looking at the finding correct way right now.

**Comment by MykolaSuperman** [comment]
> Probably because you missed `CONFIG_CMDLINE_FORCE`. But anyways, this is not a proper solution anyways. It should be configured via `xenlib`.
> 
> @Deedone is looking at the finding correct way right now.


Okay, I didn't use CONFIG_CMDLINE_FORCE because it will overwrite the u-boot bootargs parameters and we use SELinux parameters and others here, so I decided to add these parameters. In the Linux that we use for the virtual machine, this approach worked by adding CONFIG_CMDLINE_BOOL. Okay, I will wait until the correct solution is found.

**Comment by Deedone** [comment]
#21
https://github.com/aoscloud/aos_core_zephyr/pull/137

**Comment by lorc** [comment]
@MykolaSolyanko, @al1img could you please check that provided solution is working for you?

**Comment by MykolaSuperman** [comment]
> @MykolaSolyanko, @al1img could you please check that provided solution is working for you?

@lorc Thanks, I will try to check it today

**Review by mykola-kobets-epam — APPROVED** [review]

**Comment by lorc** [comment]
Closing this PR because better solution was found.

**Pull request closed without merging**